.github/workflows/validatePublicAPISurface.yml

```diff
@@ -0,0 +1,40 @@
+name: Validate Public API surface changes
+
+on:
+  workflow_dispatch:
+  push:
+  pull_request:
+    branches: [ 'main' ]
+
+permissions:
+  contents: read
+  pull-requests: write
+  issues: write
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: microsoftgraph/kiota-dom-export-diff-tool/export@main
+        id: generatePatch
+      - uses: microsoftgraph/kiota-dom-export-diff-tool/tool@main
+        if: ${{ steps.generatePatch.outputs.patchFilePath != '' }}
+        with:
+          path: ${{ steps.generatePatch.outputs.patchFilePath }}
+          fail-on-removal: true
+        id: diff
+      - uses: microsoftgraph/kiota-dom-export-diff-tool/comment@main
+        if: ${{ always() && steps.generatePatch.outputs.patchFilePath != '' && steps.diff.outputs.explanations != '' && github.event_name == 'pull_request' }}
```

Comment:
need to update to has explanations

Reply:
Resolved with https://github.com/microsoftgraph/msgraph-sdk-dotnet/pull/2626/commits/3919bd36c0f5f4ad262b422454262680822b5039